x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -0,0 +1,252 @@
+    mlServices;
+  const startServices = pick(coreStart, 'analytics', 'i18n', 'theme');
+  const datePickerDeps: DatePickerDependencies = {
+    ...pick(coreStart, ['http', 'notifications', 'theme', 'uiSettings', 'i18n']),
+    data: pluginStart.data,
+    uiSettingsKeys: UI_SETTINGS,
+    showFrozenDataTierChoice: false,
+  };
+
+  const previousRefresh = usePrevious(lastRefresh ?? 0);
+
+  useEffect(
+    function setUpJobsLoaded() {
+      async function loadJobs() {
+        try {
+          await mlJobService.loadJobsWrapper();
+          setJobsLoaded(true);
+        } catch (e) {
+          if (onError) {
+            onError(new Error(errorMessage));
+          }
+        }
+      }
+      if (isMounted.current === false) {
+        return;
+      }
+      loadJobs();
+
+      return () => {
+        isMounted.current = false;
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  useEffect(
+    function setUpSelectedJob() {
+      async function fetchSelectedJob() {
+        if (mlApiServices && selectedJobId !== undefined) {
+          try {
+            const { jobs } = await mlApiServices.getJobs({ jobId: selectedJobId });
+            const job = jobs[0];
+            setSelectedJob(job);
+          } catch (e) {
+            if (onError) {
+              onError(new Error(errorMessage));
+            }
+          }
+        }
+      }
+      if (isMounted.current === false) {
+        return;
+      }
+      fetchSelectedJob();
+    },
+    [selectedJobId, mlApiServices, onError]
+  );
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeHandler = useCallback(
+    throttle((e: { width: number; height: number }) => {
+      if (Math.abs(chartWidth - e.width) > minElemAndChartDiff) {
+        setChartWidth(e.width);
+      }
+    }, RESIZE_THROTTLE_TIME_MS),
+    [chartWidth]
+  );
+
+  const autoZoomDuration = useMemo(() => {
+    if (!selectedJob) return;
+    return mlTimeSeriesExplorerService?.getAutoZoomDuration(selectedJob);
+  }, [mlTimeSeriesExplorerService, selectedJob]);
+
+  const appStateHandler = useCallback(
+    (action: string, payload?: any) => {
+      /**
+       * Empty zoom indicates that chart hasn't been rendered yet,
+       * hence any updates prior that should replace the URL state.
+       */
+      switch (action) {
+        case APP_STATE_ACTION.SET_FORECAST_ID:
+          setSelectedForecastId(payload);
+          setZoom(undefined);
+          break;
+
+        case APP_STATE_ACTION.SET_ZOOM:
+          setZoom(payload);
+          break;
+
+        case APP_STATE_ACTION.UNSET_ZOOM:
+          setZoom(undefined);
+          break;
+      }
+    },
+
+    [setZoom, setSelectedForecastId]
+  );
+
+  // TODO: Remove data-shared-item as part of https://github.com/elastic/kibana/issues/179376>
+  return (
+    <EuiResizeObserver onResize={resizeHandler}>
```

Comment:
You need to add a `div` wrapper with `data-shared-item` here, or add this data attribute to the nested div.

Reply:
Good catch - updated in 6c61014b8a7ef4a3167a642e5f72f1938ea25c50